src/snowflake/snowpark/dataframe.py

```diff
@@ -834,8 +834,28 @@ def to_local_iterator(
             case_sensitive: A bool value which controls the case sensitivity of the fields in the
                 :class:`Row` objects returned by the ``to_local_iterator``. Defaults to ``True``.
         """
+
+        kwargs = {}
         if _emit_ast:
-            raise NotImplementedError("TODO SNOW-1672573: Support to_local_iterator.")
+            # Add an Assign node that applies SpDataframeToLocalIterator() to the input, followed by its Eval.
+            repr = self._session._ast_batch.assign()
+            expr = with_src_position(repr.expr.sp_dataframe_to_local_iterator)
+
+            debug_check_missing_ast(self._ast_id, self)
+
+            expr.id.bitfield1 = self._ast_id
+            if statement_params is not None:
+                for k, v in statement_params.items():
+                    t = expr.statement_params.add()
+                    t._1 = k
+                    t._2 = v
+            expr.block = block
+            expr.case_sensitive = case_sensitive
+
+            self._session._ast_batch.eval(repr)
+
+            # Flush the AST and encode it as part of the query.
+            _, kwargs["_dataframe_ast"] = self._session._ast_batch.flush()
```

Comment:
How is the result of the eval to be processed?  I'm assuming that is out of scope for this pr since we don't have end to end execution of the AST in place to return a result.  Is the idea we would write the results to a temp table and a local iterator implementation to stream the results?

Reply:
I'm not sure what the solution we decided on for this is. Writing the results to a temp table and having a local iterator implementation sounds like the right solution.